src/components/Tokens/TokenTable/TokenTable.tsx

```diff
@@ -48,26 +57,53 @@ function NoTokensState({ message }: { message: ReactNode }) {
   )
 }
 
-const LOADING_ROWS = Array.from({ length: 100 })
-  .fill(0)
-  .map((_item, index) => <LoadingRow key={index} />)
-
 export function LoadingTokenTable() {
   return (
-    <GridContainer>
+    <GridContainer fixedHeight={true}>
       <HeaderRow />
-      <TokenRowsContainer>{LOADING_ROWS}</TokenRowsContainer>
+      <TokenDataContainer>
+        <AutoSizer>
+          {({ height, width }) => (
+            <FixedSizeList className="List" height={height} width={width} itemCount={MAX_TOKENS_TO_LOAD} itemSize={70}>
+              {({ index, style }) => <LoadingRow style={style} key={index} />}
+            </FixedSizeList>
+          )}
+        </AutoSizer>
+      </TokenDataContainer>
     </GridContainer>
   )
 }
 
+interface TokenRowProps {
+  index: number
+  style: CSSProperties
+}
+
 export default function TokenTable() {
   const showFavorites = useAtomValue<boolean>(showFavoritesAtom)
 
   // TODO: consider moving prefetched call into app.tsx and passing it here, use a preloaded call & updated on interval every 60s
   const prefetchedTokens = usePrefetchTopTokens()
   const { loading, tokens, loadMoreTokens } = useTopTokens(prefetchedTokens)
 
+  const isItemLoaded = (index: number) => !loading && !!tokens && index < tokens.length
+
+  const Row = function TokenRow({ index, style }: TokenRowProps) {
+    const token = !!tokens && tokens[index]
+    if (!token || loading) {
+      return <LoadingRow style={style} key={index} />
+    }
+    return (
+      <LoadedRow
+        style={style}
```

Comment:
same comment re: `style` passthrough

Reply:
i think this actually needs to be named. it complains when I try to do the same spread done in LoadingRow with "Property 'style' does not exist on type 'IntrinsicAttributes &...."

Reply:
Yeah you have to let the react component know that is is a button/div etc in the prop typing with something like:

```
interface ButtonProps
    extends React.HTMLProps<HTMLButtonElement> {}
interface WrapperProps extends React.FunctionComponent<ButtonProps> { ... customProps };
```

Reply:
(copied from rebass types on their default button)

Reply:
ahh gotcha thanks, i think this latest commit should have the requested changes?